core/src/test/java/com/google/cloud/sql/core/DefaultAccessTokenSupplierTest.java

```diff
@@ -427,4 +440,48 @@ public void testGetTokenExpiration() {
                 Optional.of(new AccessToken("", past))))
         .isEqualTo(Optional.of(past));
   }
+
+  private static class Oauth2CredentialFactory implements CredentialFactory {
+    private final Credential credential;
+
+    private Oauth2CredentialFactory(Credential credential) {
+      this.credential = credential;
+    }
+
+    @Override
+    public HttpRequestInitializer create() {
+      return credential;
+    }
+  }
+
+  private static class Oauth2BadCredentialFactory implements CredentialFactory {
+    private final OAuth2Credentials credential;
+
+    private Oauth2BadCredentialFactory(OAuth2Credentials credential) {
+      this.credential = credential;
+    }
+
+    @Override
+    public HttpRequestInitializer create() {
+      return new HttpCredentialsAdapter(credential);
+    }
+  }
+
+  private static class GoogleCredentialsFactory implements CredentialFactory {
```

Comment:
This simulates the up-to-date implementation of CredentialFactory found in ApplicationDefaultCredentialFactory.